prov/sockets/src/sock_ep.c

```diff
@@ -1011,6 +1012,13 @@ static int sock_ep_getopt(fid_t fid, int level, int optname,
 		*optlen = sizeof(size_t);
 		break;
 
+	case FI_OPT_CM_DATA_SIZE:
+		if (*optlen < sizeof(size_t))
+			return -FI_ETOOSMALL;
```

Comment:
I don't see it in the man pages, but I had previously assumed that this call worked similarly to the `fi_getname` call in the sense that `optlen` would contain the size necessary to succeed if `-FI_ETOOSMALL` is returned. That's how it's implemented for the usNIC provider. Maybe we should considering amending the man page and implementing the same behavior here.